presto-main/src/test/java/com/facebook/presto/cost/TestCostCalculator.java

```diff
@@ -0,0 +1,376 @@
+                .cpu(1000 + 4000)
+                .memory(0)
+                .network(0);
+
+        assertCostEstimatedExchanges(
+                project,
+                costs,
+                stats)
+                .cpu(1000 + 4000)
+                .memory(0)
+                .network(0);
+    }
+
+    @Test
+    public void testJoin()
+    {
+        Map<Integer, PlanNodeCostEstimate> costs = ImmutableMap.of(
+                1, cpuCost(new Estimate(6000)),
+                2, cpuCost(new Estimate(1000)));
+
+        Map<Integer, PlanNodeStatsEstimate> stats = ImmutableMap.of(
+                3, statsEstimate(12000),
+                1, statsEstimate(6000),
+                2, statsEstimate(1000));
+
+        JoinNode partitionedJoin = join(3,
+                tableScan(1, "orderkey"),
+                tableScan(2, "orderkey_0"),
+                JoinNode.DistributionType.PARTITIONED,
+                "orderkey",
+                "orderkey_0");
+
+        assertCost(
+                partitionedJoin,
+                costs,
+                stats)
+                .cpu(12000 + 6000 + 1000 + 6000 + 1000)
+                .memory(1000)
+                .network(0);
+
+        assertCostEstimatedExchanges(
+                partitionedJoin,
+                costs,
+                stats)
+                .cpu(12000 + 6000 + 1000 + 6000 + 1000 + 6000 + 1000)
+                .memory(1000)
+                .network(6000 + 1000);
+
+        JoinNode replicatedJoin = join(3,
+                tableScan(1, "orderkey"),
+                tableScan(2, "orderkey_0"),
+                JoinNode.DistributionType.REPLICATED,
+                "orderkey",
+                "orderkey_0");
+
+        assertCost(
+                replicatedJoin,
+                costs,
+                stats)
+                .cpu(12000 + 6000 + 10000 + 6000 + 1000)
+                .memory(10000)
+                .network(0);
+
+        assertCostEstimatedExchanges(
+                replicatedJoin,
+                costs,
+                stats)
+                .cpu(12000 + 6000 + 10000 + 6000 + 1000)
+                .memory(10000)
+                .network(NUMBER_OF_NODES * 1000);
+    }
+
+    @Test
+    public void testAggregation()
+    {
+        Map<Integer, PlanNodeCostEstimate> costs = ImmutableMap.of(1, cpuCost(new Estimate(6000)));
+        Map<Integer, PlanNodeStatsEstimate> stats = ImmutableMap.of(
+                1, statsEstimate(6000),
+                2, statsEstimate(8));
+
+        AggregationNode aggregationNode = aggregation(2,
+                tableScan(1, "orderkey"));
+
+        assertCost(aggregationNode, costs, stats)
+                .cpu(6000 + 6000)
+                .memory(8)
+                .network(0);
+
+        assertCostEstimatedExchanges(aggregationNode, costs, stats)
+                .cpu(6000 + 6000 + 6000)
+                .memory(8)
+                .network(6000);
+    }
+
+    private CostAssertionBuilder assertCost(
```

Comment:
Use build methods from `PlanBuilder`.  End extend that class if needed. 
You can cherry-pick changes from our sprint branch as a single commit.